lib/component-utils/styles.js

```diff
@@ -0,0 +1,32 @@
+const stylesheetCache = new Map(); // key is the tag name, value is a CSSStyleSheet instance
+
+/**
+ * Apply CSS to a ShadowRoot, caching the CSS styles by the ShadowRoot host's tag name where possible.
+ * @example
+ * class MyComponent extends HTMLElement {
+ *    constructor() {
+ *      this.el = this.attachShadow({mode: `open`});
+ *      attachStaticStyles(this.el, `:host > * { background: white }`);
+ *    }
+ *  }
+ */
+export function attachStaticStyles(attachedShadowRoot, styleText) {
```

Comment:
I'm not sure I see the value of adding this whole module with stateless helper plus types file just for the sake of an extra 10 lines of component-init code. I'd rather make it an instance method or better just inline after L319 of component.js and be able to take advantage of the instance context - for instance, rather than reaching up to detect the host tagname, your cache could use the component's Class as a key and not worry about DOM logic.

Reply:
Yeah fair enough, I initially planned to reuse this for `style-override` but the logic for that ended up diverging, it makes sense to just inline it. I'll look into using the component class as cache key instead of tag name.